File: Swap Nodes in Pairs/Swap Nodes in Pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head==null||head.next==null){
            return head;
        } 
        ListNode newhead=new ListNode(0);
        newhead.next=head;
        for(ListNode temp=newhead;temp.next!=null&&temp.next.next!=null;temp=temp.next.next){
            temp.next=swap(temp.next,temp.next.next);
        }
        return newhead.next;
    }
    
    ListNode swap(ListNode first, ListNode second){
        first.next=second.next;
        second.next=first;
        return second;
    }
}

File: Swap Nodes in Pairs/Swap Nodes in Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    # @param a ListNode
    # @return a ListNode
    def swapPairs(self, head):
        newhead = ListNode(0)
        newhead.next = head
        pre=newhead
        cur =head
        while cur and cur.next:
            pre.next = cur.next
            cur.next = pre.next.next
            pre.next.next = cur
            pre = cur
            cur = cur.next
        return newhead.next
